spec/bucket_store_integration_spec.rb

```diff
@@ -182,4 +182,27 @@
   include_examples "adapter integration", "disk://bucket"
   include_examples "adapter integration", "s3://bucket"
   include_examples "adapter integration", "gs://bucket"
+
+  context "inmemory storage" do
+    let(:bucket_store) { described_class.for("#{base_bucket_uri}/#{temp_filename}") }
+    let(:base_bucket_uri) { "inmemory://bucket" }
+    let(:temp_filename) { "temp-file" }
+
+    shared_examples "encoding persistance" do |encoding|
+      it "persists encoding between upload/download for #{encoding}", :aggregate_failures do
+        upload_data = String.new("some string", encoding: encoding)
+        bucket_store.upload!(upload_data)
+        download = bucket_store.download
+        download_data = download.fetch(:content)
```

Comment:
With `let(:upload_data)` string encoding bleeds out into the next test and fails 🤷